code/processes/idb.q

```diff
@@ -0,0 +1,93 @@
+/-default parameters
+\d .idb
+
+/-these parameters are only used once their value has been set with values retrieved from the WBD.
+writedownmode:idbdir:savedir:currentpartition:symfilepath:`;
+symsize:partitionsize:0;
+
+/-force loads sym file
+loadsym:{[]
+    symfilehaschanged[];
+    .lg.o[`load;"sym file has changed, reloading"];
+    @[load;symfilepath; {.lg.e[`load;"failed to load sym file: ",string[symfilepath]," error: ",x];'x}];
+ };
+
+/-force loads IDB
+loadidb:{[]
+    partitioncounthaschanged[];
+    .lg.o[`load;"number of partitions on disk has changed, reloading"];
+    @[system; "l ", 1_string idbdir; {.lg.e[`load;"failed to load IDB: ",string[idbdir]," error: ",x];'x}];
+ };
+
+/- force loads the idb and the sym file
+loaddb:{[]
+    starttime:.proc.ct[];
+    loadsym[];
+    loadidb[];
+    .lg.o[`load;"IDB load has been finished for partition: ",string[currentpartition],". Time taken(ms): ",string .proc.ct[]-starttime];
+ };
+
+/- sets current partition and force loads the idb and the sym file. Called by the WDB after EOD.
+rollover:{[pt]
+    currentpartition::pt;
+    idbdir::.Q.dd[savedir; currentpartition];
+    loaddb[];
+ };
+
+/- reloads the db. Called by wdb process midday/eod.
+intradayreload:{[]
+    starttime:.proc.ct[];
+    if[symfilehaschanged[];loadsym[]];
+    if[partitioncounthaschanged[];loadidb[]];
```

Comment:
I think there's some redundancy here - it probably doesn't add much time etc., but we call these checks again inside the loadsym/loadidb functions. Any way we can refactor this to only do the checks once?